authentik/policies/geoip/models.py

```diff
@@ -0,0 +1,81 @@
+"""GeoIP policy"""
+
+from django.contrib.postgres.fields import ArrayField
+from django.db import models
+from django.utils.translation import gettext as _
+from rest_framework.serializers import BaseSerializer
+
+from authentik.policies.exceptions import PolicyException
+from authentik.policies.geoip.countries import COUNTRIES
+from authentik.policies.geoip.exceptions import GeoIPNotFoundException
+from authentik.policies.models import Policy
+from authentik.policies.types import PolicyRequest, PolicyResult
+
+
+class GeoIPPolicyMode(models.TextChoices):
```

Comment:
Usually we don't have this as a setting directly on the policy but instead use the `negate` flag on the policy binding to inverse the effects. For this case I'm a bit on the fence as you can do two different things in the same policy with it

Reply:
Oh, I didn't notice that flag! That makes everything so much simpler. I'm not on the fence about it at all :smile: 

Then there's no need for `*_mode` and this model will just have 2 fields: `asn_list` and `country_list`. Same as Event Matcher Policy for least astonishment:

> If any of the configured values match, the policy passes.